internal/service/enrollmentrequest.go

```diff
@@ -322,3 +327,14 @@ func (h *ServiceHandler) ReplaceEnrollmentRequestStatus(ctx context.Context, nam
 	result, err := h.store.EnrollmentRequest().UpdateStatus(ctx, orgId, &er)
 	return result, StoreErrorToApiStatus(err, false, api.EnrollmentRequestKind, &name)
 }
+
+func (s *ServiceHandler) allowCreationOrUpdate(ctx context.Context, orgId uuid.UUID, name string) bool {
+	device, err := s.store.Device().Get(ctx, orgId, name)
+	if err == nil && device != nil {
```

Comment:
The error handling pattern in golang is to handle the error first then perform additional checks. This might help your flow here as well.